Detectors/ZDC/workflow/src/DigitRecoSpec.cxx

```diff
@@ -48,136 +51,133 @@ DigitRecoSpec::DigitRecoSpec()
   mTimer.Reset();
 }
 
-DigitRecoSpec::DigitRecoSpec(const int verbosity, const bool debugOut)
-  : mVerbosity(verbosity), mDebugOut(debugOut)
+DigitRecoSpec::DigitRecoSpec(const int verbosity, const bool debugOut,
+                             const bool enableZDCTDCCorr, const bool enableZDCEnergyParam, const bool enableZDCTowerParam, const bool enableBaselineParam)
+  : mVerbosity(verbosity), mDebugOut(debugOut), mEnableZDCTDCCorr(enableZDCTDCCorr), mEnableZDCEnergyParam(enableZDCEnergyParam), mEnableZDCTowerParam(enableZDCTowerParam), mEnableBaselineParam(enableBaselineParam)
 {
   mTimer.Stop();
   mTimer.Reset();
 }
 
 void DigitRecoSpec::init(o2::framework::InitContext& ic)
 {
-  mccdbHost = ic.options().get<std::string>("ccdb-url");
+}
+
+void DigitRecoSpec::updateTimeDependentParams(ProcessingContext& pc)
+{
+  // we call these methods just to trigger finaliseCCDB callback
+  pc.inputs().get<o2::zdc::ModuleConfig*>("moduleconfig");
+  pc.inputs().get<o2::zdc::RecoConfigZDC*>("recoconfig");
+  pc.inputs().get<o2::zdc::ZDCTDCParam*>("tdccalib");
+  if (mEnableZDCTDCCorr) {
+    pc.inputs().get<o2::zdc::ZDCTDCCorr*>("tdccorr");
+  }
+  if (mEnableZDCEnergyParam) {
+    pc.inputs().get<o2::zdc::ZDCEnergyParam*>("adccalib");
+  }
+  if (mEnableZDCTowerParam) {
+    pc.inputs().get<o2::zdc::ZDCTowerParam*>("towercalib");
+  }
+  if (mEnableBaselineParam) {
+    pc.inputs().get<o2::zdc::BaselineParam*>("basecalib");
+  }
 }
 
 void DigitRecoSpec::run(ProcessingContext& pc)
 {
+  updateTimeDependentParams(pc);
   if (!mInitialized) {
     mInitialized = true;
-    // Initialization from CCDB
-    auto& mgr = o2::ccdb::BasicCCDBManager::instance();
-    mgr.setURL(mccdbHost);
-    /*long timeStamp = 0; // TIMESTAMP SHOULD NOT BE 0
-    if (timeStamp == mgr.getTimestamp()) {
-      return;
-    }
-    mgr.setTimestamp(timeStamp);*/
-
-    std::string loadedConfFiles = "Loaded ZDC configuration files for timestamp " + std::to_string(mgr.getTimestamp()) + ":";
-    auto* moduleConfig = mgr.get<o2::zdc::ModuleConfig>(o2::zdc::CCDBPathConfigModule);
-    if (!moduleConfig) {
-      LOG(fatal) << "Missing ModuleConfig ZDC configuration object";
-      return;
-    } else {
-      loadedConfFiles += " ModuleConfig";
-    }
-    if (mVerbosity > DbgZero) {
-      LOG(info) << "Loaded ZDC module configuration for timestamp " << mgr.getTimestamp();
-      moduleConfig->print();
+    std::string loadedConfFiles = "Loaded ZDC configuration files:";
+    {
+      // Module configuration
+      auto config = pc.inputs().get<o2::zdc::ModuleConfig*>("moduleconfig");
```

Comment:
and this is not needed if you are using updateTimeDependentParams / finaliseCCDB